core/src/main/java/kafka/server/share/DelayedShareFetchPartitionKey.java

```diff
@@ -18,19 +18,21 @@
 
 import kafka.server.DelayedOperationKey;
 
-import org.apache.kafka.common.Uuid;
-
 import java.util.Objects;
 
 /**
- * A key for delayed share fetch purgatory that refers to the topic partition.
+ * A key for delayed share fetch purgatory that refers to the topic partition. Since the below replicaManager functionalities
+ * use TopicPartition and not TopicIdPartition, hence we are using the same here.
```

Comment:
Hmmm, this can be problematic else we need to have deletion case handled separtely. We should not move to topic but stick to topicId.

Reply:
I agree. In general, all new code that uses topics should use topic ID.

Reply:
Agree that using topicId is probably more accurate since we could reset the state in a share partition when a topic is recreated.